servant/src/Servant/Utils/Enter.hs

```diff
@@ -39,6 +39,9 @@ instance ( Enter typ1 arg1 ret1, Enter typ2 arg2 ret2
 instance (Enter b arg ret) => Enter (a -> b) arg (a -> ret) where
     enter arg f a = enter arg (f a)
 
+instance Enter (arg -> ret) arg ret where
+    enter arg f = f arg
```

Comment:
Hm, this still seems like it muddles the purpose of `enter`. `enter` should apply a natural transformation to the return types of functions combined with `:<|>`. I don't think it should take on extra burdens, unless we can very clearly state it's new purpose (without using a conjuction or disjunction in the explanation). There are a lot of similar transformations I can imagine, and if we start including them all as instances for `enter` without a good overarching story, it'll become pretty hard to understand or explain what `enter` does.

I think I myself would prefer code that looks a lot like `enter` but just for what we want to do in the client. It's not like there'd be much duplicated code. Conceptually, if I'm getting this right, we need only two instances - one for recursing over `:<|>` and one for applying the client functions to the baseurl/manager arg.